### ISSUE TITLE
Text breaks until restart with an certain char sequence.

### ISSUE THREAD

**Issue by KalilDev** [alacritty/alacritty#4578]
So, after `cat`ing an file, my shell glitched out. I tried running `sh`, `bash` and `zsh` with the same result. I tried another terminal emulator and the tty and neither displayed the same bug as alacritty.
I then used `head` and `tail` to track the exact segment of the file, an 177 bytes segment that glitches out the terminal.

### System

OS: Linux
Version: alacritty 0.6.0 (04cbf767)
Linux/BSD: Wayland, SwayWM. Reproduced with both XWayland and native Wayland.

### Image:
[Here](https://imgur.com/VXjNb3F)

### Steps to reproduce
1. Download the following file: [File](https://ufile.io/7dm54ah9)
2. Run `cat file`
3. Done!

### Logs
"alacritty -vv" output: [Log](https://gist.github.com/KalilDev/764a3778be64681cb69fe55fb4610add)

I will later try to run alacritty with a debugger to see if i can find the exact point in the source that results in this bork.

**Comment by kchibisov** [comment]
Everything works as expected afaics. you're entering line drawing mode, not sure if a lot of terminals support it.
```
[2020-12-16 06:20:51.346916581] [TRACE] [alacritty_terminal] Configuring charset G1 as SpecialCharacterAndLineDrawing
```

In general, don't 'cat' binary files, and if you want to cat them do `cat -v`.

**Comment by kchibisov** [comment]
It's actually interesting  that I don't seem to find other terminal which supports line drawing mode. So I'll this issue open for now.

That's how you enter it: `echo -e '\e)0'`, and I'm really not sure if a lot of terminals support it.

**Comment by chrisduerr** [comment]
XTerm supports this too. This is not a bug.

**Comment by kchibisov** [comment]
Are you sure? I've tried xterm and it wasn't working in it? Maybe some settings are required?

I've decided to keep this open, since if no-one really supports it, we could remove it...

**Comment by chrisduerr** [comment]
All terminals I've tested had this working by default. Though your escape is incorrect, it should be `echo -e "\e(0"`.

**Comment by kchibisov** [comment]
Btw, could you trigger that mode in other terminals if you 'cat', since I think I won't be able to do so.

**Comment by chrisduerr** [comment]
This probably has to do with discarding bytes in the VT parser.

This is the escape responsible for setting the line drawing mode: `echo -e "\e)0\x0e"`

The escape itself seems to also activate line drawing mode in other terminals, but only Alacritty seems to get this escape from this specific file. If you `cat` a file or print the escape obviously doesn't matter.

I'll investigate some more and re-open this issue if I find out that this is indeed something that Alacritty does incorrectly and should be fixed. But the general issue exists with every terminal, it's just this specific escape that is questionable, so chances are I'll just open a separate issue on the VTE github.

**Comment by chrisduerr** [comment]
I've opened https://github.com/alacritty/alacritty/pull/4580 to fix an issue with our ESC parser.

I'd like to be very clear though that the issue reported here is **not** fixed and will never be. There are escapes that can "mess up" your terminal state and that's perfectly fine. You shouldn't cat binary files if you don't want that to happen.

This specific binary file just happened to have a sequence of bytes that exposed a bug in our parser, which is why Alacritty behaves differently here than other terminal emulators. If the escape sequence is present in the file in the same way I've posted in above (`\e)0\x0e`) every terminal will still be "messed up".
